bip-0048.mediawiki

```diff
@@ -84,7 +84,7 @@ Hardened derivation is used at this level.
 ===Script===
 
 This level splits the key space into two separate <code>script_type</code>(s). To provide
-backward and forward compatibility.
```

Comment:
I have originally added the `script_type` level for forward compatibility. Specifically to avoid a BIP44/49/84 scenario, that is, to avoid creating a new BIP every time we wanted to use new type of scripts -- instead I thought it's easier to maintain a list of script types.

from https://github.com/spesmilo/electrum/issues/4352#issue-321556261 :
>  The advantage would be that when new script types come later (e.g. Schnorr-like sigs), no new BIP would be needed.

There is another solution, which is to e.g. hash a descriptor or similar of the script type template to arrive at a derivation path level; then a single BIP would be enough; though that might make disaster-recovery problematic -- but this too requires its own `script_type` level.